src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -0,0 +1,158 @@
+        {
+            rocksdb::Options options;
+            options.create_if_missing = true;
+            rocksdb::DB* dbRawPtr;
+            const auto status {rocksdb::DB::Open(options, dbPath, &dbRawPtr)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Failed to open RocksDB database");
+            }
+            // Assigns the raw pointer to the unique_ptr. When db goes out of scope, it will automatically delete the
+            // allocated RocksDB instance.
+            m_db.reset(dbRawPtr);
+        }
+
+        /**
+         * @brief Put a key-value pair in the database.
+         *
+         * @param key Key to put.
+         * @param value Value to put.
+         *
+         * @note If the key already exists, the value will be overwritten.
+         */
+        void put(const std::string& key, const std::string& value)
+        {
+            if (key.empty() || value.empty())
+            {
+                throw std::invalid_argument("Key or value is empty");
+            }
+
+            const auto status {m_db->Put(rocksdb::WriteOptions(), key, value)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Error putting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Get a value from the database.
+         *
+         * @param key Key to get.
+         * @param value Value to get
+         */
+        void get(const std::string& key, std::string& value)
+        {
+            const auto status {m_db->Get(rocksdb::ReadOptions(), key, &value)};
+            if (status.IsNotFound())
+            {
+                throw std::invalid_argument("Key not found: " + key);
+            }
+            else if (!status.ok())
+            {
+                throw std::runtime_error("Error getting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Delete a key-value pair from the database.
+         *
+         * @param key Key to delete.
+         */
+        void delete_(const std::string& key) // NOLINT
+        {
+            const auto status {m_db->Delete(rocksdb::WriteOptions(), key)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Error deleting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Get the last key-value pair from the database.
+         *
+         * @return std::pair<std::string, std::string> Last key-value pair.
+         */
+        std::pair<std::string, std::string> getLastKeyValue()
+        {
+            std::string lastKey {};
+            std::string lastValue {};
+            std::unique_ptr<rocksdb::Iterator> it(m_db->NewIterator(rocksdb::ReadOptions()));
+            for (it->SeekToLast(); it->Valid(); it->Prev())
+            {
+                lastKey = it->key().ToString();
+                lastValue = it->value().ToString();
+                break;
+            }
```

Comment:
```suggestion
            for (it->SeekToLast(); it->Valid(); it->Prev())
            {
                lastKey = it->key().ToString();
                lastValue = it->value().ToString();
                break;
            }
```

it->SeekToLast();
if (it->Valid())
{
    lastKey = it->key().ToString();
    lastValue = it->value().ToString();
}
else
{
     throw std::runtime_error{"bla");
}

Reply:
done